src/components/table/index.tsx

```diff
@@ -1,13 +1,44 @@
-import { Component, Element, Event, EventEmitter, h, Listen } from "@stencil/core"
+import { Component, ComponentWillLoad, Element, Event, EventEmitter, h, Listen, Prop } from "@stencil/core"
 
 @Component({
 	tag: "smoothly-table",
 	styleUrl: "style.css",
 	scoped: true,
 })
-export class Table {
+export class Table implements ComponentWillLoad {
+	private owner?: EventTarget
+	private expandable: Map<EventTarget, { allowSpotlight: (allowed: boolean) => void } | undefined> = new Map()
+	private expanded: Set<EventTarget> = new Set()
 	@Element() element: HTMLSmoothlyTableElement
-	@Event() loadMore: EventEmitter<void>
+	@Prop({ mutable: true, reflect: true }) root = true
+	@Event() smoothlyNestedTable: EventEmitter<() => void>
+	@Event() spotlightChange: EventEmitter<{ allowSpotlight: boolean; owner?: EventTarget }>
+	@Event() tableLoad: EventEmitter<(owner: EventTarget) => void>
+	componentWillLoad() {
+		this.smoothlyNestedTable.emit(() => (this.root = false))
+		this.tableLoad.emit((owner: EventTarget) => (this.owner = owner))
+	}
+	@Listen("expandableLoad")
+	handleExpandableLoaded(event: CustomEvent<{ allowSpotlight: (allowed: boolean) => void }>) {
+		event.stopPropagation()
+		event.target && this.expandable.set(event.target, event.detail)
+	}
+	@Listen("expandableChange")
+	handleExpandableState(event: CustomEvent<boolean>) {
+		event.stopPropagation()
+		event.target && (event.detail ? this.expanded.add(event.target) : this.expanded.delete(event.target))
+		this.spotlightChange.emit({ allowSpotlight: !this.expanded.size, owner: this.owner })
+	}
+	@Listen("spotlightChange")
```

Comment:
why we emit an event and listen the event in the same place?

Reply:
line30

Reply:
We need to tell the parent table how to handle the nested table. Hence the need for `event.target != this.element` since if we emit the event we catch it ourselves but we want it to propagate to the next table.